merino-cache/src/memory.rs

```diff
@@ -191,6 +191,9 @@ impl Suggester {
         self.metrics_client
             .gauge("cache.memory.storage-len", self.items.len_storage() as u64)
             .ok();
+        self.metrics_client
+            .time_micros("cache.memory.duration-ms", duration)
```

Comment:
You want the plain `time` method for millisecond reporting as `time_micros` reports microseconds instead.

FYI all the timers in merino currently are measuring Provider response times. They're all basically in-memory providers at this point which return responses very quickly (sub millisecond), hence the `time_micros` usage everywhere.

However I think you're right to prefer millis here -- after populating the memory provider locally via the load tester I see the removal task taking around 2-12 ms to complete.

Reply:
Adjusting the `time` call hopefully fixes the failing test. Additionally, we might as well add a new assertion for the new metric in that same test `cache_maintainer_removes_expired_entries` -- probably a one liner

Reply:
Gotcha. Thanks for discussing. Would love to go over the testing you did locally to get that metric value. It appears I have to implement the StatsdClient differently as I cannot access the `time` variable, solution looks to be referenced in `client.rs`. I'm poking around right now, but don't want to change a major implementation detail without you feedback first. Appears to be boxed already, but somewhat differently than in the client.rs.

https://github.com/mozilla-services/merino/blob/cfdd68dc6f6461e583241bc0f0097947199d986f/merino-cache/src/memory.rs#L103 

`/// Trait that encompasses all other traits for sending metrics.
///
/// If you wish to use `StatsdClient` with a generic type or place a
/// `StatsdClient` instance behind a pointer (such as a `Box`) this will allow
/// you to reference all the implemented methods for recording metrics, while
/// using a single trait. An example of this is shown below.
///
/// ```
/// use std::time::Duration;
/// use cadence::{MetricClient, StatsdClient, NopMetricSink};
///
/// let client: Box<dyn MetricClient> = Box::new(StatsdClient::from_sink(
///     "prefix", NopMetricSink));
/// ...
/// client.time("some.timer", 42).unwrap();
/// client.time("some.timer", Duration::from_millis(42)).unwrap();

/// ````

Reply:
I think this might be a bit easier than that.

The error you might be getting is something like:
```rust
error[E0599]: no method named `time_with_tags` found for struct `StatsdClient` in the current scope
   --> merino-cache/src/memory.rs:297:22
    |
297 |                     .time_with_tags("cache.memory.duration", now.elapsed())
    |                      ^^^^^^^^^^^^^^ method not found in `StatsdClient`
    |
   ::: /home/jrconlin/.cargo/registry/src/github.com-1ecc6299db9ec823/cadence-0.26.0/src/client.rs:274:8
    |
274 |     fn time_with_tags<'a>(&'a self, key: &'a str, time: T) -> MetricBuilder<'_, '_, Timer>;
    |        -------------- the method is available for `StatsdClient` here
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
8   | use cadence::Timed;
    |

For more information about this error, try `rustc --explain E0599`.
```

Adding the `use` brings the `Timed` trait in context. This helps the compiler find the right methods it needs so you can do 

```rust
                self.metrics_client
                    .time_with_tags("cache.memory.duration", now.elapsed())
                    .with_tag("cache-status", response.cache_status.to_string().as_str())
                    .send();
```
that should let you compile without significant changes. 

(Mind you, locally I'm getting odd internal actix build errors which I think might be related to a version conflict triggering inside of the common-rs library. Hopefully, you're not getting that.)

Reply:
Thanks @jrconlin , I was getting some build errors that were different, but mostly due to the M1 architecture on my laptop which related to something not building locally, but that was solved when committed and pushed. But yes, the addition of the use statement allowed the use of `time` which can then be ingested by StatsdClient, without having to add the extra tag.